dask/array/core.py

```diff
@@ -1573,9 +1573,489 @@ def __float__(self):
+                    stop = size
+                    if index.stop < loc1:
+                        stop -= loc1 - index.stop
+
+                    start = index.start - loc0
+                    if start < 0:
+                        # Make start positive
+                        start %= index.step
+
+                    if start >= stop:
+                        # This block does not overlap the indices
+                        overlaps = False
+                        break
+
+                    # Still here? Then this block does overlap the
+                    # indices
+                    step = index.step
+                    block_index = slice(start, stop, step)
+                    block_index_size, rem = divmod(stop - start, step)
+                    if rem:
+                        block_index_size += 1
+
+                    # Find how many elements of 'value' precede this
+                    # block along this dimension. Note that it is
+                    # assumed that the slice step is positive, as will
+                    # be the case for reformatted indices.
+                    pre = index.indices(loc0)
+                    n_preceeding, rem = divmod(pre[1] - pre[0], step)
+                    if rem:
+                        n_preceeding += 1
+                else:
+                    # Index is a list of integers
+                    block_index = [i - loc0 for i in index if loc0 <= i < loc1]
+                    if not block_index:
+                        # This block does not overlap the indices
+                        overlaps = False
+                        break
+
+                    # Still here? Then this block does overlap the
+                    # indices
+                    block_index_size = len(block_index)
+                    if block_index_size == 1:
+                        block_index = slice(block_index[0], block_index[0] + 1)
+                    else:
+                        index0 = block_index[0]
+                        step = block_index[1] - index0
+                        if step > 0:
+                            start, stop = index0, block_index[-1] + 1
+                        elif step < 0:
+                            start, stop = index0, block_index[-1] - 1
+
+                        if block_index == list(range(start, stop, step)):
+                            # Replace the list of integers with a slice object
+                            if stop < 0:
+                                stop = None
+
+                            block_index = slice(start, stop, step)
+
+                    # Find how many elements of 'value' precede this
+                    # block along this dimension. Note that it is
+                    # assumed that the sequence is strictly
+                    # monotonically increasing, as will be the case
+                    # for reformatted indices.
+                    n_preceeding = sum(1 for i in index if i < loc0)
+                # --- End: if
```

Comment:
Showing where the if ends seems out of place in dask.

Reply:
Sure - I like them on particularly large or indented clauses, but I know that's unusual! They've gone.